### DISCUSSION TITLE
Controlling width of Options Picker

### DISCUSSION THREAD

**Discussion by hefferbub** [Budibase/budibase#8596] [Help] [answered]
I'm new to the platform so forgive me if this is obvious.

I have a small form I want to display near the top of a screen which displays and allows users to change some default values.  These values are persisted through State variables, not a data provider.

In the Builder, I can set the widths of these Pickers and that is reflected correctly.  However, when I preview, the Pickers get compressed down to minimum size.  See images.

I have tried everything I can think of, but am still stumped.  What am I missing or misunderstanding?

Thanks
<img width="972" alt="Screenshot 2022-11-08 at 11 13 57 AM" src="https://user-images.githubusercontent.com/1672667/200619227-9a2e3589-4ab3-44ed-a3a7-bcedb7444d71.png">
<img width="1681" alt="Screenshot 2022-11-08 at 11 13 04 AM" src="https://user-images.githubusercontent.com/1672667/200619231-f5dc2aeb-f2fe-4e62-a6f8-3b3b4be1a1f0.png">

**Comment by hefferbub** [comment]
I think this may be a (small) bug.  

I recreated these elements in a new empty form and they act just the same--sizing works fine in the Builder but it shrinks to minimum when displayed in Preview.

What is unusual about this is that the fields are not linked to any data source as they normally would be.  Instead, I use them to maintain the value of some State variables.  I load the picker Options for the two fields from different data tables, but the value of the Fields are "State.Default Entity" and "State.Default Budget Year".

I imagine this is an odd usage, but it seems to work aside from the sizing problem.

Any ideas to try or alternative suggestions?

Thanks
<img width="1338" alt="Screenshot 2022-11-09 at 3 22 15 PM" src="https://user-images.githubusercontent.com/1672667/200935737-3d2b3d02-702f-45bc-b684-a2f10e130ff9.png">
<img width="703" alt="Screenshot 2022-11-09 at 3 21 06 PM" src="https://user-images.githubusercontent.com/1672667/200935739-e4cee62e-6395-4c84-bd31-d7041ce7cf5b.png">

**Comment by Overness** [comment]
I confirm that I face the same problem.

**Comment by amit-meshram** [comment]
I am facing same issue. Did someone find a solution to this problem?

**Comment by dadlMilestone** [comment]
I haven't found a solution, but a workaround. I use two nested containers with the options picker inside the inner container. 

The outer container is set to direction horizontal, the inner two containers are set to default with the size set the maximize. Then the options picker is place into the inside container. Here is a screenshot that will hopefully help:

![image](https://user-images.githubusercontent.com/110244200/214663981-d37e29e2-717b-475d-8f8c-1cfe83a66163.png)

**Comment by melohagan** [comment, marked as answer]
Hey @hefferbub @Overness @dadlMilestone @amit-meshram 

This is actually a minor bug: https://github.com/Budibase/budibase/issues/8750

Within the builder you need to provide the unit, e.g. `250px` and `500px`, etc.

Apologies for the late response, I'm only noticing this discussion today!
